nservicebus/handlers/accessing-data.md

```diff
@@ -0,0 +1,67 @@
+
+snippet:BusinessData-Native-NotIdempotent
+
+Should the handler be invoked more than once (e.g. because of a transient problem with the transport while committing the receive transaction) the data will get corrupted (duplicate rows will be inserted). The following code shows how to mitigate the problem:
+
+snippet:BusinessData-Native-Idempotent
+
+The downside of this approach is the fact that the code gets more complex. Sometimes it is hard to get enough information from the incoming message to create correct *idempotent* handling logic. This is especially true for commands. 
+
+### Transport using distributed transactions
+
+When the selected transport is configured to use distributed transactions via Distributed Transaction Coordinator (DTC) service, the handler is executed within an ambient transaction scope. If the data store supports enlisting in a distributed transaction (e.g. SQL Server, Oracle), the state changes are guaranteed to be applied successfully exactly once, atomically with the message receive operation.
+
+
+## Using NServiceBus persistence
+
+Instead of managing everything themselves, users can delegate the management of data store to NServiceBus persistence. This approach has a number of advantages:
+ * NServiceBus guarantees best practices are followed when it comes to managing data store's connection
+ * Data access context is automatically shared between all handlers executed for a given message, making it easier to guarantee *idempotency* (no partial successes where one handler managed to commit the changes while other didn't)
+ * Data access context is also shared with the [Saga](/nservicebus/sagas) that might participate in handling a given message
+
+The limitation of this approach is that it requires NServiceBus support for a given data store. Out of the box supported stores include SQL Server and Oracle (via NServiceBus.NHibernate persistence) and RavenDB. 
+
+partial:api
+
+Consult the documentation of a particular persistence for details.
+
+ * [NHibernate](/nservicebus/nhibernate/accessing-data.md)
+ * [RavenDB](/nservicebus/ravendb/#shared-session)
+
+### Transport using native transactions
+
+In this mode the NServiceBus-managed data store context can be committed multiple times for a single message and it is up to the user to guarantee *idempotency*. Unlike in the basic example, though, the data store context is shared between the handlers so there might be one (possibly generic) handler that takes care of the *idempotency*, allowing others to focus on pure business problem.
+
+snippet:BusinessData-Native-Managed
+
+
+### Outbox
+
+When using the [Outbox](/nservicebus/outbox) NServiceBus itself guarantees the *idempotency* of data access operations executed via `SynchronizedStorageSession`. The Outbox can be seen as a generic implementation of the *idempotency*-ensuring handler from the previous section.
```

Comment:
SynchronizedStorageSession is defined in a partial, earlier in this doco. I switch to partials for outbox because SynchronizedStorageSession  is only available in 6.